packages/ui/src/components/styled-link.tsx

```diff
@@ -1,17 +1,23 @@
 import { Link, type LinkProps } from 'react-router-dom'
 
 import { cn } from '@utils/cn'
+import { cva, VariantProps } from 'class-variance-authority'
 
-type StyledLinkProps = LinkProps & React.RefAttributes<HTMLAnchorElement>
+const linkVariants = cva(
+  'whitespace-nowrap underline decoration-transparent decoration-1 underline-offset-4 transition-colors duration-200',
+  {
+    variants: {
+      variant: {
+        default: 'text-foreground-1 hover:decoration-foreground-1',
+        secondary: 'text-foreground-4 hover:decoration-foreground-4',
+        accent: 'text-foreground-accent hover:decoration-foreground-accent'
+      }
+    }
+  }
+)
 
-export function StyledLink({ className, ...props }: StyledLinkProps) {
-  return (
-    <Link
-      className={cn(
-        'text-foreground-accent hover:decoration-foreground-accent underline decoration-transparent underline-offset-4 transition-colors duration-200',
-        className
-      )}
-      {...props}
-    />
-  )
+export type StyledLinkProps = LinkProps & React.RefAttributes<HTMLAnchorElement> & VariantProps<typeof linkVariants>
```

Comment:
Can we import `RefAttributes`? `React` isn't imported, so this triggers a lint warning.